src/test/fuzz/process_message.cpp

```diff
@@ -0,0 +1,236 @@
+        m_node_context.banman = MakeUnique<BanMan>(GetDataDir() / "banlist.dat", nullptr, DEFAULT_MISBEHAVING_BANTIME);
+        m_node_context.connman = MakeUnique<CConnman>(0x1337, 0x1337);
+        m_thread_group.create_thread(std::bind(&CScheduler::serviceQueue, &m_scheduler));
+        GetMainSignals().RegisterBackgroundSignalScheduler(m_scheduler);
+        ChainstateActive().InitCoinsDB(1 << 23, true, false);
+        ChainstateActive().InitCoinsCache();
+        if (!LoadGenesisBlock(Params())) {
+            throw std::runtime_error("LoadGenesisBlock(...) failed.");
+        }
+        BlockValidationState block_validation_state;
+        if (!ActivateBestChain(block_validation_state, Params())) {
+            throw std::runtime_error("ActivateBestChain(...) failed.");
+        }
+        for (int i = 0; i < 2 * COINBASE_MATURITY; i++) {
+            CreateAndProcessNextBlock(*m_node_context.mempool);
+            SleepUntilSchedulerCompletion(m_scheduler);
+        }
+        assert(ChainActive().Tip() != nullptr);
+        m_peer_logic_validation = MakeUnique<PeerLogicValidation>(m_node_context.connman.get(), nullptr, m_scheduler);
+        m_peer_logic_validation->InitializeNode(m_dummy_p2p_node.get());
+        assert(GetMainSignals().CallbacksPending() == 0);
+        SleepUntilSchedulerCompletion(m_scheduler, 1);
+        LogInstance().m_print_to_console = true;
+    }
+
+    ~FuzzingSetup()
+    {
+        LogInstance().DisconnectTestLogger();
+        fs::remove_all(m_path);
+        m_thread_group.interrupt_all();
+        m_thread_group.join_all();
+        GetMainSignals().FlushBackgroundCallbacks();
+        GetMainSignals().UnregisterBackgroundSignalScheduler();
+    }
+};
+
+const std::map<std::string, std::set<std::string>> EXPECTED_DESERIALIZATION_EXCEPTIONS = {
+    {"CDataStream::read(): end of data: iostream error", {"addr", "block", "blocktxn", "cmpctblock", "feefilter", "filteradd", "filterload", "getblocks", "getblocktxn", "getdata", "getheaders", "headers", "inv", "notfound", "ping", "sendcmpct", "tx"}},
+    {"index overflowed 16 bits: iostream error", {"getblocktxn"}},
+    {"index overflowed 16-bits: iostream error", {"cmpctblock"}},
+    {"indexes overflowed 16 bits: iostream error", {"getblocktxn"}},
+    {"non-canonical ReadCompactSize(): iostream error", {"addr", "block", "blocktxn", "cmpctblock", "filteradd", "filterload", "getblocks", "getblocktxn", "getdata", "getheaders", "headers", "inv", "notfound", "tx"}},
+    {"ReadCompactSize(): size too large: iostream error", {"addr", "block", "blocktxn", "cmpctblock", "filteradd", "filterload", "getblocks", "getblocktxn", "getdata", "getheaders", "headers", "inv", "notfound", "tx"}},
+    {"Superfluous witness record: iostream error", {"block", "blocktxn", "cmpctblock", "tx"}},
+    {"Unknown transaction optional data: iostream error", {"block", "blocktxn", "cmpctblock", "tx"}},
+};
+
+std::unique_ptr<FuzzingSetup> fuzzing_setup;
```

Comment:
```suggestion
std::unique_ptr<FuzzingSetup> g_fuzzing_setup;
```

This should be prefixed with `g_` to convey this is global state that is shared between fuzz calls and might lead to non-determinism.

Reply:
Good point! Fixed!